e2e/internal/upgrade/upgrade_test.go

```diff
@@ -125,6 +124,20 @@ func TestUpgrade(t *testing.T) {
 	testNodesEventuallyHaveVersion(t, k, targetVersions, *wantControl+*wantWorker, *timeout)
 }
 
+func runUpgradeCheck(require *require.Assertions, cli string) {
+	cmd := exec.CommandContext(context.Background(), cli, "upgrade", "check", "--debug")
+	msg, err := cmd.CombinedOutput()
+	require.NoErrorf(err, "%s", string(msg))
+
+	// We can't check for the exact output as it depends on multiple factors.
+	// But each CI run should at least include these lines.
+	require.True(strings.Contains(string(msg), "The following updates are available with this CLI:"))
```

Comment:
```suggestion
	require.Contains(string(msg), "The following updates are available with this CLI:")
	require.Contains(string(msg), "Kubernetes:")
	require.Contains(string(msg), "Services:")
```

Couldn't we use the test variables from above (lines 38ff) to also check for the versions the CLI wants to upgrade from/to?

Reply:
We can't check images because targetImages may be unpublished. Added checks for the other two cases.